src/entities/swap/swaps/v2/auraBalSwaps/parseInputs.ts

```diff
@@ -0,0 +1,46 @@
+import { Address } from 'viem';
+import { Token } from '@/entities/token';
+import { AuraBalSwapKind, SwapQueryInput } from './types';
+import { supportedTokens, auraBalToken } from './constants';
+import { SwapKind } from '@/types';
+import { TokenAmount } from '@/entities/tokenAmount';
+
+export interface AuraBalSwapQueryInput {
+    kind: AuraBalSwapKind;
+    swapToken: Token;
+    inputAmount: TokenAmount;
+}
+
+export function isAuraBalSwap(input: SwapQueryInput): boolean {
+    const { tokenIn, tokenOut, kind } = input;
+    return (
+        kind === SwapKind.GivenIn &&
+        tokenIn.chainId === 1 &&
+        tokenOut.chainId === 1 &&
+        (auraBalToken.isSameAddress(tokenIn.address) ||
+            auraBalToken.isSameAddress(tokenOut.address)) &&
+        (isSupportedToken(tokenIn) || isSupportedToken(tokenOut))
+    );
+}
+
+export function parseInputs(input: SwapQueryInput): AuraBalSwapQueryInput {
+    const { tokenIn, tokenOut, swapAmount } = input;
+    if (
+        !isAuraBalSwap(input) ||
+        !tokenIn.isSameAddress(swapAmount.token.address)
+    )
+        throw Error('Not A Valid AuraBal Swap');
```

Comment:
Not sure if this matters much, but you might consider modularizing some of the validation checks to throw slightly more specific error messages that hint _why_ the attempted swap input is invalid